controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -445,13 +445,10 @@ public String getScopeName() {
     @Override
     public CompletableFuture<Boolean> scaleTryDeleteEpoch(final int epoch) {
         return getHistoryTableFromStore()
-                .thenCompose(historyTable -> getSegmentTableFromStore().thenApply(segmentTable -> new ImmutablePair<>(historyTable, segmentTable)))
-                .thenCompose(pair -> {
-                    Data<T> segmentTable = pair.getRight();
-                    Data<T> historyTable = pair.getLeft();
+                .thenCompose(historyTable -> {
                     CompletableFuture<Boolean> result = new CompletableFuture<>();
 
-                    if (TableHelper.isScaleOngoing(historyTable.getData(), segmentTable.getData())) {
+                    if (TableHelper.isNewEpochCreated(historyTable.getData())) {
```

Comment:
Say new epoch is *not* created, but the commit/abort processing that leads us here is for the last txn of the epoch. How do we make sure that the old epoch is actually deleted eventually?

Reply:
tryCompleteScale method is called from 3 places:
1. scale -> after new epoch is created
2. txn commit 
3. txn abort

if commit/abort dont find new epoch created and remove txns from under old epoch and leave, scale will create new epoch, find old epoch node empty and delete old epoch.